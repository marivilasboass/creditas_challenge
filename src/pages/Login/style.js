import styled from 'styled-components/native';
import {Dimensions} from 'react-native';
import {theme} from '~/theme/globalStyle';

export const Wrapper = styled.View`
  flex: 1;
  margin-top: 20%;
  background: ${theme.color.white};
  align-items: center;
`;

export const Logo = styled.Image`
  width: ${Dimensions.get('window').width / 2};
  height: ${Dimensions.get('window').width / 2};
  margin-bottom: ${theme.defaultPadding.large};
  resize-mode: contain;
`;

export const FormWrapper = styled.KeyboardAvoidingView`
  width: 100%;
  padding-left: ${theme.defaultPadding.large};
  padding-right: ${theme.defaultPadding.large};
  padding-bottom: ${theme.defaultPadding.large};
`;

export const LoginText = styled.Text`
  font-size: ${theme.fontSize.regular};
  margin-top: ${theme.defaultPadding.small};
  margin-bottom: ${theme.defaultPadding.small};
  color: ${theme.color.black};
`;

export const Input = styled.TextInput.attrs({
  placeholderTextColor: theme.color.white,
  underlineColorAndroid: 'transparent',
})`
  width: 100%;
  height: 40px;
  margin-top: ${theme.defaultPadding.small};
  margin-bottom: ${theme.defaultPadding.small};
  padding: 0px 15px;
  border: 1px solid #ccc;
  background-color: ${theme.color.white};
  text-align: center;
`;
