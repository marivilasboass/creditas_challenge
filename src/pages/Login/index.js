import React, {useState, useEffect} from 'react';
import {useSelector, useDispatch} from 'react-redux';
import {Platform} from 'react-native';
import {Types, userLogin, userAddEmail, userReset} from '~/store/ducks/user';

import {Wrapper, Logo, FormWrapper, Input, LoginText} from './style';
import Button from '~/components/Button';

const Login = ({navigation}) => {
  const userState = useSelector(state => state.user);
  const dispatch = useDispatch();
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [isFocussed, setFocus] = useState(false);
  const keyboardBehavior = Platform.OS === 'android' ? 'padding' : 'position';

  /*
  onClickNext = () => userAddEmail({dispatch})({email});
  onClickBack = () => {
    dispatch({type: Types.LOGIN_RESET});
    setFocus(false);
  };
  onClickLogin = () =>
    userLogin({dispatch})({email: userState.email, password, navigation});
  onFocus = () => setFocus(true);
  onBlur = () => setFocus(false);
  */

  return (
    <Wrapper>
      <Logo source={require('~/assets/images/logo.png')} />
      <FormWrapper behavior={keyboardBehavior}>
        {userState.email ? (
          <>
            <LoginText>{userState.email}</LoginText>

            <Input
              placeholder="password"
              onChange={value => setPassword(value.nativeEvent.text)}
              secureTextEntry
              onFocus={this.onFocus}
              onBlur={this.onBlur}
              autoFocus
            />
            <Button title="login" onPress={this.onClickLogin} />
            <Button title="back" onPress={this.onClickBack} />
          </>
        ) : (
          <>
            <Input
              placeholder="login"
              onChange={value => setEmail(value.nativeEvent.text)}
              keyboardType="email-address"
              autoCapitalize="none"
              autoCompleteType="off"
              autoCorrect={false}
              onFocus={this.onFocus}
              onBlur={this.onBlur}
            />
            <Button title="next" onPress={this.onClickNext} />
          </>
        )}
      </FormWrapper>
    </Wrapper>
  );
};

export default Login;
