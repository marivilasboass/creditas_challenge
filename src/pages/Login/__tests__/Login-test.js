import 'react-native';
import React from 'react';
import Login from '~/pages/Login';
import renderer from 'react-test-renderer';

it('renders Login correctly (snapshot)', () => {
  const component = renderer.create(<Login />).toJSON();
  expect(component).toMatchSnapshot();
});
