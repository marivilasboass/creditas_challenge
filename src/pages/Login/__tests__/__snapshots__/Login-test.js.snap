// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Login correctly (snapshot) 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#FFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "marginTop": 20,
        "paddingLeft": 40,
        "paddingRight": 40,
      },
    ]
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/Assets/github-logo.png",
      }
    }
    style={
      Array [
        Object {
          "alignSelf": "center",
          "height": 375,
          "marginBottom": 40,
          "resizeMode": "contain",
          "width": 375,
        },
        Object {
          "opacity": 0.8,
          "top": 0,
          "transform": Array [
            Object {
              "scale": 0.3,
            },
          ],
        },
      ]
    }
  />
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "paddingBottom": 20,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "bottom": 0,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCompleteType="off"
        autoCorrect={false}
        keyboardType="email-address"
        onChange={[Function]}
        placeholder="Your Github email address"
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "backgroundColor": "#FFF",
              "borderColor": "#CCC",
              "borderWidth": 1,
              "height": 40,
              "marginBottom": 40,
              "marginTop": 40,
              "textAlign": "center",
              "width": "100%",
            },
          ]
        }
        underlineColorAndroid="transparent"
      />
      <View
        accessible={true}
        focusable={false}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "marginBottom": 20,
            "marginLeft": 20,
            "marginRight": 20,
            "marginTop": 20,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "#FFF",
              "borderColor": "#333",
              "borderRadius": 6,
              "borderWidth": 1,
              "color": "#333",
              "padding": 8,
            }
          }
        >
          <Text>
            Next
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
